app/models/id_card_attributes.rb

```diff
@@ -47,16 +47,32 @@ def title38_status_code
   end
 
   def branches_of_service
-    branches = @user.military_information.service_episodes_by_date.map do |ep|
-      SERVICE_KEYS[ep.branch_of_service_code]
-    end
+    branches = if Flipper.enabled?(:military_information_vaprofile)
+                 military_info.service_episodes_by_date.map do |ep|
+                   SERVICE_KEYS[ep.branch_of_service_code]
+                 end
+               else
+                 @user.military_information.service_episodes_by_date.map do |ep|
+                   SERVICE_KEYS[ep.branch_of_service_code]
+                 end
+               end
     branches.compact.join(',')
   end
 
   def discharge_types
-    discharges = @user.military_information
-                      .service_episodes_by_date
-                      .map(&:discharge_character_of_service_code)
+    discharges = if Settings.vsp_environment == 'production' || !Flipper.enabled?(:military_information_vaprofile)
```

Comment:
@bosawt We added this extra layer of protection to prevent the `discharge_character_of_service_code` bug from getting into production for the VIC form. 

@little-oddball has given his approval to merge this PR this morning. We wanted these changes merged before an 11am ET meeting with VA folks.